app/assets/javascripts/petition-widget-builder.js

```diff
@@ -0,0 +1,103 @@
+(function($) {
+
+  // This is not good, but I would not like to make a request to fetch this HTML
+  var template = [
+    "<div class='petition-widget-builder'>",
+    "  <h2> Incorporar petição </h2>",
+    "  <i class='close icon-close'></i>",
+    "  <div class='separator'><span></span></div>",
+    "  <p> Incorpore a petição \"<span class='petition-name'></span>\" em seu site </p>",
+    "  <h3> Personalizar </h3>",
+    "  <ul class='list-unstyled'>",
+    "    <li>",
+    "      <input type='checkbox' class='show-description' checked='checked' /><label> Exibir descrição </label>",
+    "    </li>",
+    "    <li>",
+    "      <input type='checkbox' class='show-progress' checked='checked' /><label> Exibir detalhes do progresso da votação </label>",
+    "    </li>",
+    "    <li>",
+    "      <input type='checkbox' class='show-signers' checked='checked' /><label> Exibir assinantes </label>",
+    "    </li>",
+    "  </ul>", 
+    "  <h3> Código </h3>",
+    "  <input type='text' class='source-code' readonly='readonly' />",
+    "  <div class='previewer'>",
+    "</div>"
+  ].join("");
+
+  var buildUrl = function(petitionId) {
+    return [
+      location.protocol, "//", location.host, "/embedded/petitions/", petitionId
+    ].join("");
+  };
+
+  var buildFlags = function(hasDescription, hasProgress, hasSigners) {
+    var flags = [];
+
+    if (hasDescription) flags.push("has_description");
+    if (hasProgress) flags.push("has_progress");
+    if (hasSigners) flags.push("has_signers");
+
+    return flags.join(" ");
+  };
+
+  $.fn.petitionWidgetBuilder = function(opts) {
+    opts = opts || {};
+
+    var petitionId = opts.petitionId;
+    var petitionName = opts.petitionName;
+    var cycleColor = opts.cycleColor;
+
+    if (!petitionId || !petitionName) {
+      throw "petitionId and petitionName is required"
+    }
+
+    $(this).each(function() {
+      var $element = $(this);
+      $element.hide();
+
+      var $petitionWidgetBuilder = $(template);
+      $element.append($petitionWidgetBuilder);
+
+      $element.find(".close").click(function() {
```

Comment:
I usually prefer the `#on` syntax. This is just my opinion though, as there is no diference here.

```javascript
$element.on("click", ".close", function() {})
```